app/(dashboard)/student.tsx

```diff
@@ -34,26 +45,98 @@ export default function StudentPage() {
+  );
+  const [filteredCourses, setFilteredCourses] = useState(courses);
+  const [notFocusesdcourses, setNotFocusesdcourses] = useState(courses);
+
+  const handleChange = (event: SelectChangeEvent<String>) => {
+    const selectedSort = event.target.value as String;
+    setSortBy(selectedSort);
+    setShowAllCourses(false);
+
+    // Update filteredCourses based on the selected sorting criteria
+    if (selectedSort === "title") {
+      // Sort courses alphabetically by title
+      setFilteredCourses(
+        [...courses].sort((a, b) => a.title.localeCompare(b.title))
+      );
+    } else if (selectedSort === "startYear") {
+      // Filter courses that match the currentYear
+      const filtered = courses.filter(
+        (course) => course.startYear === currentYear
+      );
+      filtered.sort((a, b) => b.startDate - a.startDate);
+      setFilteredCourses(filtered);
+      const notFocused = courses.filter(
+        (course) => course.startYear !== currentYear
+      );
+      notFocused.sort((a, b) => b.startDate - a.startDate);
+      setNotFocusesdcourses(notFocused);
+    } /* else if (selectedSort === "yearDivision") {
+      // Filter and sort courses with specific yearDivision criteria (e.g., FIRST_SEMESTER or SECOND_SEMESTER)
+      const filtered = courses.filter(
+        (course) =>
+          course.yearDivision === "FIRST_SEMESTER" &&
+          course.yearDivision !== null
+      );
+      filtered.sort((a, b) => b.startDate - a.startDate);
+      setFilteredCourses(filtered);
+
+      const notFocused = courses.filter(
+        (course) =>
+          course.yearDivision === "SECOND_SEMESTER" &&
+          course.yearDivision !== null
+      );
+      notFocused.sort((a, b) => b.startDate - a.startDate);
+      setNotFocusesdcourses(notFocused);
+    } */
+  };
+
+  const handleButton = () => {
+    setShowAllCourses(!showAllCourses);
+  };
 
   return (
     <main>
       <Typography variant="h1" gutterBottom>
         Dashboard
       </Typography>
+      {/* Sort by Selection */}
+      <Box sx={{ minWidth: 120, maxWidth: 150, float: "right" }}>
+        <FormControl fullWidth>
+          <InputLabel>Sort by</InputLabel>
+          <Select value={sortBy} onChange={handleChange} label={"Sort By"}>
+            <MenuItem value={"title"}>Alphabetically</MenuItem>
+            <MenuItem value={"startYear"}>Year</MenuItem>
+            {/* <MenuItem value={"yearDivison"}>Year Division</MenuItem> */}
+          </Select>
+        </FormControl>
+      </Box>
       <Typography variant="h2" gutterBottom>
         My Courses
       </Typography>
 
+      {sortBy === "startYear" && (
+        <Typography variant="h6" gutterBottom>
+          Current Year: {currentYear}
+        </Typography>
+      )}
+
       {/* MOCK */}
-      {courses.length === 0 ? (
+      {filteredCourses.length === 0 ? (
         <div className="text-center text-2xl text-slate-400 my-80">
           You have not joined any courses yet. Visit the{" "}
```

Comment:
"... courses for this year."

Reply:
Don't understand what you mean?

Reply:
It is possible to see this message while having joined courses (if they are all from previous years). I would change the message slightly to clarify this